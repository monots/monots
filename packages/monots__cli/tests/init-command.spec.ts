import { createSetupFixtures } from '@monots/test';
import { afterAll, test } from 'vitest';

import { cli, context } from './test-setup.js';

const setupFixtures = createSetupFixtures({ context, fileUrl: import.meta.url });

afterAll(async () => {
  await setupFixtures.cleanup();
});

test.concurrent('`monots init` should run without error in a pnpm project', async ({ expect }) => {
  const { loadJsonFile, context } = await setupFixtures('init-pnpm-with-package-json');
  const result = await cli.run(['init'], context);
  const json = await loadJsonFile('package.json');

  expect(result, 'exit with 0 when running `monots init`').toBe(0);
  expect(json).toMatchSnapshot();
});

test.concurrent('`monots init` errors when no package.json file available', async ({ expect }) => {
  const { loadJsonFile, context } = await setupFixtures('init-without-package-json');
  const result = await cli.run(['init'], context);

  expect(result, 'The result should be a failure').toBe(1);
  await expect(loadJsonFile('package.json'), 'No package.json file created').rejects.toThrow();
});

test.concurrent('`monots init` fails outside a workspace', async ({ expect }) => {
  const { loadJsonFile, context } = await setupFixtures('init-without-workspace');
  const result = await cli.run(['init'], context);
  const json = await loadJsonFile('package.json');

  expect(result, 'The result should be a failure').toBe(1);
  expect(json, 'The package.json file should be untouched').toEqual({});
});
