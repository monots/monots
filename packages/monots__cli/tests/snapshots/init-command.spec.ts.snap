// Vitest Snapshot v1

exports[`\`monots init\` should run without error in a pnpm project 1`] = `
{
  "dependencies": {
    "@monots/cli": "^0.0.0",
    "typescript": "^4.7.2",
  },
  "monots": {
    "packages": [
      "packages/*",
    ],
  },
  "scripts": {
    "postinstall": "monots prepare",
  },
  "type": "module",
}
`;
